### DISCUSSION TITLE
linking error on sample.c

### DISCUSSION THREAD

**Discussion by fernandoherrerav** [microsoft/msquic#2818] [Q&A] [answered]
Hello, If copy sample.c from msquic/src/tools/sample to for example /tmp/ and then I do:
`gcc sample.c -lmsquic`
I get the linking error on the image:

![linker](https://user-images.githubusercontent.com/83671990/173373881-d0a167fb-9510-49fa-a26a-427f1308a5be.png)

Interestingly if I run the compilation command:
`gcc sample.c -lmsquic -O1`
It works fine. It seems to be related to this: [https://stackoverflow.com/questions/16245521/c99-inline-function-in-c-file/16245669#16245669](url)

Is this behavior expected? What command do I need to run to compile sample.c outside the msquic repo?

**Comment by nibanks** [comment, marked as answer]
I don't have a good solution for you. Internally, we have the following file that we include to get around these annoying issues: https://github.com/microsoft/msquic/blob/main/src/platform/inline.c.
